### ISSUE TITLE
Allow providers to have multiple locations?

### ISSUE THREAD

**Issue by cecilia-donnelly** [SolutionGuidance/psm#413]
We allow individual providers to be associated with multiple groups at different locations.  Do we allow group providers to have multiple locations?

@brainwane has pointed out:

>I'm looking at the organizational enrollment workflow. In step 1, Organization Info: "Do not list additional practice location addresses here. If there are multiple clinic locations, each one must be enrolled separately"

So, we have some conflicting requirements around this (we have a requirement that just says "The PSM shall allow multiple locations for the provider.").  @brainwane and I had talked in IRC about possibly linking different locations together by state Medicaid ID (#347).

I'll tag this as a question for states.

**Comment by cecilia-donnelly** [comment]
This is becoming clearer, if more complicated:

- it's more the provider providing info, what services/specialties they have at a given location, not necessarily the fact that the locations are at different addresses
- [state employee]: if the provider has 5 offices, they might want to keep records separate and bill separately, but that's up to them
- @kfogel, afterward, suggested adding dates to all provider information -- yes, they're at this location on Mondays, but that was only true in 2016 or something.
- e.g. a doctor's specialties may vary by location because of available facilities, etc.
  - at home, GP, while at another practice they're practicing a specialty

So for each location we need to store their specialt(y)(ies), whether they're linked to a group practice at that location, and the start (and possibly end) dates of them practicing there.

**Comment by chj124** [comment]
1) there should still be one master provider record and all elements related to the provider must reference that 1 provider record. status on the main record is basically Active/Suspended/Inactive. Status for Provider/Facility level would have more settings. If applying again in the future, the same provider record should be used, and basic info updated. Application data should be in a separate table and a new application would be needed when re-applying.
2) data related to the provider that can be in different facilities and needs to be tracked separately would require a separate table
3) schedule of what is available where and when is definitely needed for other processes, like case management, but im not sure how detailed you want to be here. location/specialty by day of week would be nice, but i would never expect a dr/office to keep a separate system updated with the dr schedule. Especially when dr's how provide services at hospitals do not always have the same schedule.

pretty much everything needs effective dating and and a full audit trail of changes.

**Comment by cecilia-donnelly** [comment]
@brainwane, @jasonaowen, and I talked about this in IRC a bit, and came up with the framework of tracking "facilities" and "individual providers" and the "links" between those entities.  We need to track all of these enrollments and links and the associated dates:

 - something like facilty enrolls, provider enrolls, and then either side can create a connection
 - how much validation do the states want at each step?
 - [facility enrollment, provider enrollment] have the automated validation steps we already have implemented (e.g., LEIE, DMF)
 - there'd need to be a lookup of whether that facility is licensed to provide X service, if a provider creates a link claiming to be offering service X (presumably this is an automated check of some state db)
 - and some kind of confirmation to the non-initiating side (does the facility/provider accept the friend request?)
 - sole practitioners aren't necessarily linked to a facility (they may have a private practice)
 

@chj124, one thing that I want to triple-check is the status of a dental clinic, a hospital, a pharmacy, or a similar facility.  That entity (facility) will enroll as itself, as a "group provider", and then link to the individual providers that provide services at that facility.  Is that consistent with how you understand this?

What about a corporate entity that has multiple locations?  Does each of those locations enroll separately as a facility/group, or do they need to be linked together somehow?

**Comment by chj124** [comment]
i think there are some misunderstandings here....
facilities and groups do no apply or enroll. Even if they enter all the information the individual providers apply, are approved and therefore enrolled.
One provider in a group could apply for, and be approved to provide services, but its not the group or the facility that is directly approved.
Groups whether a corporate group, billing group, group practice, facility, etc… are just for reference as far as were concerned. But we do need to track the NPI number of each group.
For that matter, the facility class in the attached class diagram could be for facility too.
if a provider does brain surgery at facility Y, the provider applies, as part on the application they say they can perform brain surgery, they provide the proper credentials, and specify where the brain surgery will occur. as part of reviewing the providers application, a risk assessment will be performed and brain surgery is assigned a high risk. so a site visit will be scheduled to see that the facility is appropriate. after the site visit, and all approvals, the provider will be enrolled to perform brain surgery at the specified facility for a certain time period for the specific plan (Medicaid).
as long as the provider bills for brain surgery, for an enrolled client, and a prior authorization was approved, his bill will be accepted.
so this means that when a provider is approved/enrolled for a service at a facility, then the facility is approved for that service by association.
Also, when a provider is approved/enrolled for a service at a facility, then the group is also approved by association.
A group of whatever type can have multiple providers, and one or more may be approved and enrolled, but not necessarily all of them. They are still individually vetted.
The group itself has no credentials to prove it can provide any services, that come from the providers.
Bottom line, everyone still needs to be tracked (groups, facilities, providers) but not everyone applies and possible gets approved/enrolled.

it would be a preference of each state whether an facility/group approval is needed so the ability should be there. Also long as their registered, the NPI is properly represented, the providers approval is key.

![ss1](https://user-images.githubusercontent.com/5740678/31461559-e2894ee8-ae97-11e7-8eaf-651b06e2cbf7.png)

the simplified version could be a better implementation

![ss2](https://user-images.githubusercontent.com/5740678/31462241-1803919e-ae9a-11e7-8f57-97630478eae1.png)
